code/game/machinery/vending.dm

```diff
@@ -444,26 +444,12 @@
 		return 0
 	else
 		// Okay to move the money at this point
-		var/paid = customer_account.charge(currently_vending.price, GLOB.vendor_account,
```

Comment:
Where will the sale be credited? And what happens if payment fails?

Reply:
Payment is credited to the Vendor account. Previous checks ensure that the payment will  be go through, but if it somehow managed to fail anyway the user would still receive the item.